src/main/java/de/hpi/bpt/argos/parsing/staticData/StaticDataParserImpl.java

```diff
@@ -0,0 +1,277 @@
+		 *                         .
+		 *                         .
+		 *                     </attributes>
+		 *                     <childInstances>
+		 *                         .
+		 *                         .
+		 *                         .
+		 *                     </childInstances>
+		 *                 </instance>
+		 *                 		.
+		 *                 		.
+		 *                 		.
+		 *             </childInstances>
+		 *         </instance>
+		 *         		.
+		 *         		.
+		 *         		.
+		 *     </instances>
+		 * </staticData>
+		 */
+	}
+
+	/**
+	 * This method returns the singleton instance of this class.
+	 * @return - the singleton instance of this class
+	 */
+	public static StaticDataParser getInstance() {
+		if (instance == null) {
+			instance = new StaticDataParserImpl();
+		}
+
+		return instance;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void loadStaticData() {
+		if (!Argos.shouldLoadStaticData()) {
+			return;
+		}
+
+		getArtifactBatch().setup();
+
+		String staticDataDirectory = Argos.getStaticDataDirectory();
+		if (staticDataDirectory.length() == 0) {
+			logger.info("static data directory not defined in properties");
+			return;
+		}
+
+		// since the path is delivered as URI, it is represented as a HTML string. Thus we need to replace %20 with file system spaces.
+		File dataDirectory = new File(staticDataDirectory.replaceAll("%20", " "));
+
+		if (!dataDirectory.isDirectory()) {
+			logger.error("static data directory is not a directory");
+			return;
+		}
+
+		File[] dataFiles = dataDirectory.listFiles();
+
+		if (dataFiles == null) {
+			logger.info("no static data files to parse");
+			return;
+		}
+
+		for (File staticDataFile : dataFiles) {
+			if (!staticDataFile.getName().endsWith(".xml")) {
+				continue;
+			}
+
+			WatchTask parseFileTask = null;
+
+			try {
+				int oldEntityTypesCount = PersistenceAdapterImpl.getInstance().getEntityTypesCount();
+				int oldEntitiesCount = PersistenceAdapterImpl.getInstance().getEntitiesCount();
+
+				parseFileTask = WatchImpl.start(String.format("parsing static data file: '%1$s'", staticDataFile.getName()),
+						() -> parse(staticDataFile));
+
+				int newEntityTypesCount = PersistenceAdapterImpl.getInstance().getEntityTypesCount();
+				int newEntitiesCount = PersistenceAdapterImpl.getInstance().getEntitiesCount();
+
+				logger.info(String.format("'%1$s' contained [%2$d entityTypes] and [%3$d entities]",
+						staticDataFile.getName(),
+						newEntityTypesCount - oldEntityTypesCount,
+						newEntitiesCount - oldEntitiesCount));
+
+			} catch (Exception e) {
+				LoggerUtilImpl.getInstance().error(logger, String.format("cannot parse '%1$s'", staticDataFile.getName()), e);
+
+			} finally {
+				if (parseFileTask != null) {
+					parseFileTask.finish();
+				}
+			}
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void startElement(String elementName) {
+		switch (elementName) {
+			case ENTITY_TYPES_ROOT_ELEMENT:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1151)